common/src/app/common/files/validate.cljc

```diff
@@ -123,11 +123,17 @@
 
         (doseq [child-id (:shapes shape)]
           (let [child (ctst/get-shape page child-id)]
-            (when (or (nil? child) (not= (:parent-id child) (:id shape)))
+            (if (nil? child)
               (report-error! :child-not-found
                 (str/ffmt "Child % not found" child-id)
-                child file page
-                :parent-id (:id shape)))))))))
+                shape file page
+                :parent-id (:id shape)
+                :child-id child-id)
+              (when (not= (:parent-id child) (:id shape))
+                (report-error! :invalid-parent
+                  (str/ffmt "Invalid parent %" child-id)
```

Comment:
In this trace it would be useful to show both the child-id and the parent-id of the child

Reply:
For example `(str/ffmt "Child %s has invalid parent %2 child-id parent-id)`

Reply:
I think you have made the change in the wrong line :smile:

Reply:
oh!, sorry!